freud/util.pyx

```diff
@@ -212,8 +212,12 @@ def _convert_array(array, shape=None, dtype=np.float32):
     Returns:
         :class:`numpy.ndarray`: Array.
     """
+    if array is None:
+        return np.empty(shape, dtype=dtype)
```

Comment:
We should specify in the docstring that if `array is None`, the calling code is responsible for setting the initial data. Otherwise it is possible that the caller might not realize that and expect an array of zeros.